astroplan/core.py

```diff
@@ -0,0 +1,275 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+#from ..extern import six
+
+import sys
+#from math import sqrt, pi, exp, log, floor
+#from abc import ABCMeta, abstractmethod
+
+import numpy as np
+
+# Originally authored by Jazmin Berlanga Medina (jazmin.berlanga@gmail.com),
+# and modified by Adrian Price-Whelan (email) and Erik Tollerud (email).
+
+__all__ = ["Observer", "Target", "FixedTarget", "NonFixedTarget",
+           "Constraints", "Observation"]
+
+#__doctest_requires__ = {'*': ['scipy.integrate']}
+
+
+class Observer(object):
+    """
+    Some comments.
+    """
+
+    def __init__(self, name, longitude, latitude, elevation, pressure,
+                 rel_humidity, temp, timezone, description=None):
+        """
+        Comments.
+        """
+        raise NotImplementedError()
+
+    def set_environment():
+        """
+        Comments.
+        """
+        raise NotImplementedError()
+
+    # Methods to 
+
+    def get_date(self):
+        """
+        Compatible with astropy.time.Time.
+
+        Must first create date object with get_date() to then feed to all other methods below.
+        """
+        raise NotImplementedError()
+
+    # Sun-related methods.
+
+    def noon(Date):
```

Comment:
This is just a silly convention thing: in most Python code, capitalizing a first letter indicates a class name whereas lowercase first letter indicates an instantiated object -- here you probably want to use `date` or `datetime` (well, actually, see comment above, you'll need `(self, datetime)`.